app/controllers/data_sources_controller.rb

```diff
@@ -82,6 +82,15 @@ def destroy
     redirect_to action: :index
   end
 
+  def authoritative_types
```

Comment:
Let's move this to `DataSource`, then it can be used to show a human read-able authoritative type on the show page.